kpc/views.py

```diff
@@ -138,21 +141,40 @@ def test_func(self):
 
 
 class CertificateView(BaseCertificateView):
+    REVIEW_MSG = "Please review the certificate data below."
 
     def get_form_class(self):
         if self.object.licensee_editable:
             return LicenseeCertificateForm
         return StatusUpdateForm
 
+    def get_form(self):
+        """
+        Check for incoming GET params
+        If present, populate form with values
+        """
+        form = super().get_form()
+        if self.request.GET and isinstance(form, LicenseeCertificateForm):
+            messages.info(self.request, self.REVIEW_MSG)
+            return LicenseeCertificateForm(self.request.GET)
+        return form
+
     def get_template_names(self):
         if self.object.licensee_editable:
             return ['certificate/details-edit.html']
         return ['certificate/details.html']
 
     def form_valid(self, form):
-        response = super().form_valid(form)
-        messages.success(self.request, form.SUCCESS_MSG)
-        return response
+        """Render confirmation page if certificate submitted"""
+        if isinstance(form, LicenseeCertificateForm):
+            cert = form.save(commit=False)
+            context = self.get_context_data()
+            context['b64_pdf'] = CertificatePreview(cert).make_preview()
```

Comment:
This is a little awkward, as we're injecting template vars outside of `get_context_data`. Is there a way to rejigger it to keep context in that function?